### DISCUSSION TITLE
help with the map component and leaflet

### DISCUSSION THREAD

**Discussion by d3liaa** [noi-techpark/it.bz.opendatahub.databrowser#175] [Q&A]
I'm trying to create a map component using leaflet and vue2-leaflet. 
But I encountered an issue immediately at the beginning trying to install vue2-leaflet on the data browser directory.
This is the error: 
![Immagine 2022-10-24 155824](https://user-images.githubusercontent.com/101118017/197545003-e4ce49b7-ebd0-435d-8a7c-5b9ab1584982.png)
I would kindly appreciate any insight and help :)

**Comment by gappc** [comment]
Hi @d3liaa, I think the problem is that you try to use a library that was made for Vue 2, while the databrowser uses Vue 3. The error message says:

`"Could not resolve dependency vue@"^2.5.17" from vue2-leaflet@2.7.1"`

Please try to find an alternative library (e.g. https://github.com/vue-leaflet/vue-leaflet) and check if that works. Let me know if the problems persist.

**Reply by d3liaa** [reply]
Thank you! I will try to use this now :))

**Reply by d3liaa** [reply]
Apparently, it doesn't support typescript, so i tried to create a module declaration with a file .d.ts how they described the issue . But i've never done it before so I'm not sure how it works.
https://github.com/vue-leaflet/vue-leaflet/issues/1

**Comment by gappc** [comment]
@d3liaa Thank you for taking a look at it. The last entry in the thread you linked points to OpenLayers as alternative solution to Leaflet. Could you please take a look at that solution and see if / how it works? There seems to be a lib for Vue 3: https://github.com/MelihAltintas/vue3-openlayers
